extensions/stats/plugin.cc

```diff
@@ -170,12 +170,18 @@ const wasm::common::NodeInfo& NodeInfoCache::getPeerById(
     LOG_INFO(absl::StrCat("cleaned cache, new cache_size:", cache_.size()));
   }
 
-  google::protobuf::Struct metadata;
-  if (getMetadataStruct(MetadataType::Request, peer_metadata_key, &metadata) !=
-      Common::Wasm::WasmResult::Ok) {
+  auto metadata_value =
+      getSelectorExpression({"filter_state", peer_metadata_key});
+  if (!metadata_value.has_value()) {
     LOG_DEBUG(absl::StrCat("cannot get metadata for: ", peer_metadata_key));
     return cache_[""];
   }
+  google::protobuf::Struct metadata;
+  if (!metadata.ParseFromArray(metadata_value.value()->data(),
```

Comment:
It seems this is common enough that we should just extract this into a function and put it in  common context?

Reply:
Right, updated to use `getStructValue`. Messy to juggle multiple repositories.